### PR TITLE
CSI: allow kadalu_format 'native' or 'non-native' in all config

### PR DESCRIPTION
**Pull request by amarts**
* also remove '.external' part from storageclass name for external
  as it doesn't make sense anymore after merging #589

Signed-off-by: Amar Tumballi <amar@kadalu.io>

### CODE REVIEW

**Comment by amarts** [comment]
@vatsa287 with this, if you handle `kadalu_format` check in CreateVolume in even non `External` type volumes, we can resolve #587 much smoothly.

@aravindavk what do you think.

**Review by leelavg — COMMENTED** [review]
- Removing 'external' keyword is a good idea
- IMHO, it's better to have kadalu_format only for external storage as kadalu storages by default use subdir
- Even if we consider Heketi, I think those will fall under external non-native 🤔 

Nevertheless, if we still want to have kadalu_format better we have only one option at `spec` level in CRD and remove it from `details`. In addition to that, below things should be taken care
1. https://github.com/kadalu/kadalu/blob/devel/cli/kubectl_kadalu/storage_add.py#L121
2. https://github.com/kadalu/kadalu/blob/devel/cli/kubectl_kadalu/storage_add.py#L277
3. https://github.com/kadalu/kadalu/blob/devel/operator/main.py#L492

**Comment by amarts** [comment]
> Even if we consider Heketi, I think those will fall under external non-native thinking 

It makes sense to provide kadalu native way for migration, that way, it would become users' choice to use it either way. (We already have option of external 1-1 PV).

**Review by aravindavk — APPROVED** [review]
LGTM

**Review by vatsa287 — APPROVED** [review]

File: doc/external-gluster-storage.adoc [hunk 1]
```diff
@@ -31,14 +31,14 @@ metadata:
   name: ext-config
 spec:
   type: External
+  # Omitting 'kadalu_format' or using 'native' as a value will create
```
Comment (vatsa287) [inline doc/external-gluster-storage.adoc:34]: Will not kadalu_format: non-native create 1PV = 1Subdir? So shouldn't value above should be non-native?

**Pull request merged**